src/main/java/frc/robot/hardware/mechanisms/wpilib/ElevatorSimulation.java

```diff
@@ -0,0 +1,45 @@
+package frc.robot.hardware.mechanisms.wpilib;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.simulation.ElevatorSim;
+import frc.utils.Conversions;
+import frc.utils.time.TimeUtils;
+
+public class ElevatorSimulation implements WPILibMechanismSimulation {
+
+	private final ElevatorSim elevatorSimulation;
+	private final double wheelDiameter;
+	private final double gearRatio;
+
+	public ElevatorSimulation(ElevatorSim elevatorSimulation, double wheelDiameter, double gearRatio) {
+		this.elevatorSimulation = elevatorSimulation;
+		this.wheelDiameter = wheelDiameter;
+		this.gearRatio = gearRatio;
+	}
+
+	@Override
+	public Rotation2d getSystemPosition() {
+		return Conversions.distanceToAngle(elevatorSimulation.getPositionMeters(), wheelDiameter);
+	}
+
+	@Override
+	public Rotation2d getSystemVelocityAnglesPerSecond() {
```

Comment:
AnglesPreSecond? Rename to rotationsPerSecond

Reply:
@YHerm